types/index.d.ts

```diff
@@ -63,8 +63,8 @@ declare module "fido2-library" {
   }
 
   interface ExpectedAttestationResult {
-    rpId: string;
-    origin?: string | string[];
+    rpId?: string;
+    origin: string;
```

Comment:
This looks like a breaking change to me

Reply:
well this one was just adjustment according to js code, e.g. if you tried to provide origin as string array it would throw an error on validation, that's why I adjusted it to be just a string. Do you think that I should restore it?

Reply:
I was mainly talking about the question mark - if I remember correctly this means that before origin was optional to provide, while now it isnt. This might break some scripts, wouldn't it?

Reply:
Actually origin isn't optional in latest package, it's required and type must be string. If you try to omit it in latest version (master branch) of package or provide it as string array then it would throw an error.
On the other side in latest version if you try to provide rpId value it would throw an error, so index.d.ts is actually outdated based on js codebase in master branch

Reply:
examples from master branch:

```
  const expectedReg = {
    challenge: "LN6BmfUMFzBgfmqoaAU26k2Ef7STFrZ5eHF8P_q309usd_Vj-qMurZaKLVvW60OE31l_3jKucjQ_1kA0mp5aNA",
    origin: ["https://example.com"],
    factor: "either"
  }
  regResponse.id = Buffer.from(regResponse.id, 'base64').buffer

  const regRes = await f2l.attestationResult(regResponse, expectedReg)
  console.log(true)

# throws:
TypeError: expected 'origin' should be string, got object
    at Object.parseExpectations (...../playground/node_modules/fido2-library/lib/parser.js:26:10)
```

```
  const expectedReg = {
    challenge: "LN6BmfUMFzBgfmqoaAU26k2Ef7STFrZ5eHF8P_q309usd_Vj-qMurZaKLVvW60OE31l_3jKucjQ_1kA0mp5aNA",
    rpId: "example.com",
    factor: "either"
  }
  regResponse.id = Buffer.from(regResponse.id, 'base64').buffer

  const regRes = await f2l.attestationResult(regResponse, expectedReg)
  console.log(true)

# throws:
Error: expectation did not contain value for 'origin'
    at Fido2AttestationResult.validateExpectations (..../playground/node_modules/fido2-library/lib/validator.js:38:10)
```

Reply:
As you can see from examples above if we make requests according to original index.d.ts we would get errors since it doesn't comply with actual js code:
```ts
  interface ExpectedAttestationResult {
    rpId: string;
    origin?: string | string[];
    challenge: string;
    factor: Factor;
  }
```

Reply:
I see - thanks for the good explanation!